crates/typst/src/syntax/parser.rs

```diff
@@ -306,14 +312,38 @@ fn math_expr_prec(p: &mut Parser, min_prec: usize, stop: SyntaxKind) {
         p.wrap(m, SyntaxKind::Math);
     }
 
+    let mut primed = false;
+
     while !p.eof() && !p.at(stop) {
+        // if `stop` is hat, than prime is also stop
+        // a_b' works like a_b^'
+        if stop == SyntaxKind::Hat && p.at(SyntaxKind::Prime) {
+            break;
+        }
+
         if p.directly_at(SyntaxKind::Text) && p.current_text() == "!" {
             p.eat();
             p.wrap(m, SyntaxKind::Math);
             continue;
         }
 
+        if p.eat_if(SyntaxKind::Prime) {
+            // eat as many primes as possible
+            while p.eat_if(SyntaxKind::Prime) {}
```

Comment:
this will also group primes with spaces in between. do we want that?

Reply:
I agree. The typically used construct is p.directly_at. There is not direct alternative to eat_if at the moment.